src/client_side_request.cc

```diff
@@ -1637,6 +1637,37 @@ ClientHttpRequest::loggingEntry(StoreEntry *newEntry)
         loggingEntry_->lock("ClientHttpRequest::loggingEntry");
 }
 
+void
+ClientHttpRequest::initRequest(HttpRequest *aRequest)
+{
+    assert(aRequest);
+    assert(!request);
+    request = aRequest;
+    HTTPMSGLOCK(request);
+    if (ConnStateData *csd = getConn()) {
+        if (!csd->connectionTag().isEmpty()) {
+            if (!request->notes)
+                request->notes = new NotePairs;
+            // TODO: assert if "clt_conn_tag" already added?
+            request->notes->add("clt_conn_tag", SBuf(csd->connectionTag()).c_str());
```

Comment:
What happens to the adapted request? Will it miss this `clt_conn_tag`? Or is there some code that will copy that tag from virgin to the adapted request?

Reply:
At least in icap code the notes are copied via HttpRequest::inheritProperties() (ModXact.cc).